### PR TITLE
Feature/set acl

### PR DESCRIPTION
**Pull request by chasinandrew**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [X] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [X] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import com.google.cloud.resourcemanager.v3.ProjectsClient;
+import com.google.iam.v1.Binding;
+import com.google.iam.v1.Policy;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+public class SetAcl {
+  public static void setAcl() throws IOException,
```
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java:37]: This should be main method.

File: content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    /* TODO(developer): Replace these variables before running the samples
+    * Find ACL roles here https://cloud.google.com/document-warehouse/docs/roles-and-permissions */
+    String projectId = "your-project-id"; 
+    String location = "your-region"; // Format is "us" or "eu"
+    String userId = "your-userid"; // Format is user:xxxx
+    ArrayList<String> members = new ArrayList<>(); // Format is ["user:xxxx", "user:xxxx"]
+    String role = "your-role";        
+  }
+
+  public static void setAcl(String projectId, String location,
```
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java:48]: Kindly add a method comment.

File: content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      endpoint = String.format("%s-%s", location, endpoint);
+    }
+    DocumentServiceSettings documentServiceSettings =
+        DocumentServiceSettings.newBuilder().setEndpoint(endpoint).build();
+        
+    // Define Request Metadata for enforcing access control
+    RequestMetadata requestMetadata = RequestMetadata.newBuilder()
+        .setUserInfo(
+        UserInfo.newBuilder()
+            .setId(userId).build()).build();
```
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java:63]:
```suggestion
            .setId(userId).build())
            .build();
```

File: content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        
+    /*  Define IAM binding. This will add a list of members to be bound to a role.
+    *   Please see the below documentation for IAM predefined roles for content warehouse.
+    *   https://cloud.google.com/document-warehouse/docs/roles-and-permissions */ 
+    Binding binding = Binding.newBuilder()
+        .addAllMembers(members)
+        .setRole(role).build();
```
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java:70]:
```suggestion
        .setRole(role)
        .build();
```

File: content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        setAclResponse = documentServiceClient.setAcl(setAclRequest);
+
+      } else { 
+        SetAclRequest setAclRequest = SetAclRequest.newBuilder()
+            .setResource(String.format("%s/%s", "projects", projectNumber))
+            .setRequestMetadata(requestMetadata)
+            .setPolicy(aclPolicy)
+            .setProjectOwner(true).build();
+
+        setAclResponse = documentServiceClient.setAcl(setAclRequest);
```
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java:98]: This line can be moved out of the if..else loop.

File: content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        /* The full resource name of the document, e.g.: 
+        projects/{project_number/locations/{location/documents/{document_id} */
+        DocumentName documentName = 
+            DocumentName.ofProjectLocationDocumentName(projectNumber, location, documentId);
+        SetAclRequest setAclRequest = SetAclRequest.newBuilder()
+            .setResource(documentName.toString())
+            .setRequestMetadata(requestMetadata)
+            .setPolicy(aclPolicy).build();
+
+        setAclResponse = documentServiceClient.setAcl(setAclRequest);
```
Comment (Sita04) [inline content-warehouse/src/main/java/contentwarehouse/v1/SetAcl.java:89]:
```suggestion
        
```

File: content-warehouse/src/test/java/contentwarehouse/v1/SetAclTest.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class SetAclTest {
+  @Rule public final MultipleAttemptsRule multipleAttemptsRule = new MultipleAttemptsRule(3);
+ 
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String LOCATION = "us";
+  private static final String USER_ID = "user:andrewchasin@google.com";
+  private static final String DOCUMENT_ID = "06rubracospg8";
```
Comment (Sita04) [inline content-warehouse/src/test/java/contentwarehouse/v1/SetAclTest.java:43]: Kindly remove these values from here and include it as part of the CI env vars.

File: content-warehouse/src/test/java/contentwarehouse/v1/SetAclTest.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  }
+   
+  @Before
+  public void setUp() {
+    MEMBERS.add(0, "user:andrewchasin@google.com");
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    originalPrintStream = System.out;
+    System.setOut(out);
+  }
```
Comment (Sita04) [inline content-warehouse/src/test/java/contentwarehouse/v1/SetAclTest.java:70]:
```suggestion
  
  @BeforeAll
  public static void setup() {
    requireEnvVar("GOOGLE_CLOUD_PROJECT");
    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
    MEMBERS.add(0, USER_ID);
  }
   
  @Before
  public void setUp() {
    bout = new ByteArrayOutputStream();
    out = new PrintStream(bout);
    originalPrintStream = System.out;
    System.setOut(out);
  }
```

Checking required env variables and adding the User ID should be part of the `BeforeAll` annotation, which executes exactly once before the tests are run.
Including it as part of the`Before` annotation would make them run before every test is executed.

File: content-warehouse/src/test/java/contentwarehouse/v1/SetAclTest.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+@RunWith(JUnit4.class)
+public class SetAclTest {
+  @Rule public final MultipleAttemptsRule multipleAttemptsRule = new MultipleAttemptsRule(3);
+ 
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String LOCATION = "us";
+  private static final String USER_ID = "user:andrewchasin@google.com";
+  private static final String DOCUMENT_ID = "06rubracospg8";
+  private static final ArrayList<String> MEMBERS = new ArrayList<>();
```
Comment (Sita04) [inline content-warehouse/src/test/java/contentwarehouse/v1/SetAclTest.java:44]:
```suggestion
  private static ArrayList<String> MEMBERS = new ArrayList<>();
```

File: content-warehouse/src/test/java/contentwarehouse/v1/SetAclTest.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    assertThat(got).contains("policy");
+  }
+   
+  @After
+  public void tearDown() {
+    System.out.flush();
+    System.setOut(originalPrintStream);
+  }
+}
+
```
Comment (Sita04) [inline content-warehouse/src/test/java/contentwarehouse/v1/SetAclTest.java:86]: Include a new line here.

**Review by Sita04 — CHANGES_REQUESTED** [review]
Thanks for the sample. PTAL at the comments and also the failing tests.

**Comment by chasinandrew** [comment]
Closing this PR as this sample is deprioritized

**Pull request closed without merging**